src/scripts/main.js

```diff
@@ -1,3 +1,38 @@
 'use strict';
 
-// write code here
+const bodyElement = document.body;
+
+const dataSalary = [...bodyElement.querySelectorAll('li[data-salary]')];
+const listPeople = [...bodyElement.querySelectorAll('li')];
+
+const salaries = dataSalary
+  .map((elem) => +elem.getAttribute('data-salary').replace(/[$,]+/g, ''))
+  .sort((a, b) => b - a);
+
+const sortedArray = [];
+
+function sortList(list) {
+  for (let i = 0; i < list.length; i++) {
+    const salary = salaries[i];
+
+    const person = list.filter(
+      (elem) =>
+        +elem.getAttribute('data-salary').replace(/[$,]+/g, '') === salary,
```

Comment:
The comparison '+elem.getAttribute('data-salary').replace(/[$,]+/g, '') === salary' may not work as expected because 'salary' is a number and the result of the expression is a string. Consider converting the result of the expression to a number before comparing.